### PR TITLE
AHTI-202 AHTI-203 | Import harbors from venepaikka

### PR DESCRIPTION
**Pull request by charn**
This PR adds venepaikka harbour data importer. Harbours are imported as Features without any harbour specific data.

### CODE REVIEW

File: features/importers/myhelsinki_places/tests/test_import_images.py [hunk 1]
```diff
@@ -106,10 +107,13 @@ def test_image_is_deleted(requests_mock, importer, places_response):
 
     Most likely it means that the image not available on the previous url.
     """
+    ImageFactory(
```
Comment (igordavydsson) [inline features/importers/myhelsinki_places/tests/test_import_images.py]: what about the test above, i.e. `test_delete_image_with_wrong_new_license`? it also runs `importer.import_features()` twice.

Comment (charn) [reply features/importers/myhelsinki_places/tests/test_import_images.py]: Fair point, I'll refactor

File: features/importers/venepaikka_harbors/config.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "tag_config": {"id": "ahti:tag:island", "name": "saaristo"},
+  "category_config": {"id": "ahti:category:island", "name": "Saaret"},
```
Comment (igordavydsson) [inline features/importers/venepaikka_harbors/config.json]: shouldn't these be:
```
"tag_config": {"id": "ahti:tag:harbor", "name": "satama"},
"category_config": {"id": "ahti:category:harbor", "name": "Satamat"}
```
?

File: features/importers/venepaikka_harbors/importer.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+    def import_features(self):
+        source_type = self.get_source_type()
+        client = VenepaikkaHarborsClient()
+        harbors = client.fetch_harbors(query=query).json()
+
+        for harbor in feature_expression.search(harbors):
+            feature = self._import_feature(harbor, source_type)
+            self._import_feature_category(feature, app_settings.CATEGORY_CONFIG)
+            self._import_feature_tag(feature, app_settings.TAG_CONFIG)
```
Comment (igordavydsson) [inline features/importers/venepaikka_harbors/importer.py]: if all of the features get the same tag and category, wouldn't it be DRYer to create `Category` and `Tag` objects before this loop and then assign them to all the features?

File: features/importers/venepaikka_harbors/importer.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+            "geometry": Point(harbor["lon"], harbor["lat"], srid=settings.DEFAULT_SRID),
+            "source_type": st,
+        }
+        feature, created = Feature.objects.language("fi").update_or_create(
+            source_type=st, source_id=harbor["id"], defaults=values,
+        )
+        return feature
+
+    def _import_feature_category(self, feature: Feature, category_config: Mapping):
+        """Imports and set category for the given Feature.
```
Comment (igordavydsson) [inline features/importers/venepaikka_harbors/importer.py]: `... and setS category ...` 👀

File: features/importers/venepaikka_harbors/tests/test_import_harbors.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+
+def test_source_type_gets_created(requests_mock, importer):
+    empty_response = {"data": {"harbors": {"edges": []}}}
+    requests_mock.post(HARBORS_URL, json=empty_response)
+
+    importer.import_features()
+
+    assert SourceType.objects.count() == 1
+    SourceType.objects.get(system="venepaikka", type="harbor")
```
Comment (igordavydsson) [inline features/importers/venepaikka_harbors/tests/test_import_harbors.py]:
```python
assert SourceType.objects.filter(system="venepaikka", type="harbor").exists()
```
?

Comment (charn) [reply features/importers/venepaikka_harbors/tests/test_import_harbors.py]: I could be more explicit here that would definitely be more readable. :+1: But as a sidenote, if the get fails DoesNotExist will be raised and the test will fail.

File: features/importers/venepaikka_harbors/config.json (hunk 2 above)
Comment (charn) [reply features/importers/venepaikka_harbors/config.json]: They should. :sob: Looks like some version of copy&paste error.

File: features/importers/venepaikka_harbors/tests/test_import_harbors.py (hunk 5 above)
Comment (igordavydsson) [reply features/importers/venepaikka_harbors/tests/test_import_harbors.py]: yep, I meant that would it be more readale with `assert ... exists()` 😅

File: features/importers/venepaikka_harbors/importer.py [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+            self._set_feature_tag(feature, tag)
+            self._import_feature_contact_info(feature, harbor["address"])
+            self._import_service_map_url(feature, harbor["servicemap_id"])
+            self._import_feature_images(feature, harbor["images"], image_license)
+
+    @staticmethod
+    def _import_feature(harbor: dict, st: SourceType) -> Feature:
+        values = {
+            "name": harbor["name"],
+            "url": harbor["url"],
```
Comment (igordavydsson) [inline features/importers/venepaikka_harbors/importer.py]: so, will you keep this imported for now?

Comment (charn) [reply features/importers/venepaikka_harbors/importer.py]: Ah, that.. I'll remove it for now.

File: features/importers/venepaikka_harbors/importer.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    },
+    images: [properties.imageFile, properties.imageLink]
+    servicemap_id: properties.servicemapId
+}
+"""
+)
+
+
+class VenepaikkaImporter(FeatureImporterBase):
+
```
Comment (igordavydsson) [inline features/importers/venepaikka_harbors/importer.py]: so, do you prefer to have empty lines before attributes' declarations in a class or not? differs between classes within this commit 👀

Comment (charn) [reply features/importers/venepaikka_harbors/importer.py]: Same issue also on the `MyHelsinkiImporter`. But no worries, a fix is on the way. :wrench:

**Pull request merged**